src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/ClientErrorResultFilter.cs

```diff
@@ -43,6 +43,11 @@ public void OnResultExecuting(ResultExecutingContext context)
                 return;
             }
 
+            if (clientError.StatusCode < 400)
```

Comment:
Looking at this code I realise that maybe we got the name wrong... I would expect it to be possible to use `ProblemDetails` for a 500, and you can. Maybe it's a minor detail. 

Perhaps worth a comment that we're intentionally not putting an upper bound on this, we want to allow folks to translate a 5XX.